### DISCUSSION TITLE
Help with installing Azuracast on Synology Docker

### DISCUSSION THREAD

**Discussion by boer025** [AzuraCast/AzuraCast#3874] [Q&A]
I'm searching for a step-by-step guide for installing Azuracast on a Synology Docker. 
Is there already such a thing, or someone who can help me with that?

Thanks in advance.

**Comment by Radio-Quergedacht** [comment]
https://www.youtube.com/watch?v=ITepoNvdF5o

**Comment by Radio-Quergedacht** [comment]
https://www.azuracast.com/install/

**Reply by ghost** [reply]
page not found for this URL

**Comment by monigrip** [comment]
Looking for the same install guide for docker synology. I'm assuming I've port conflicts at this stage. 80 - 443.

**Comment by monigrip** [comment]
> I'm searching for a step-by-step guide for installing Azuracast on a Synology Docker.
> Is there already such a thing, or someone who can help me with that?
> 
> Thanks in advance.

I've managed to install it on 1019+. You'll need to ssh into you diskstation to do some of these things. Use the docker cli, ./docker.sh install. Disable any VPN. I had to delete the first few range of ports for the station container (8000-8016) on the .yml file within the /var/Azuracast folder. as I had a conflict. After the install go to your ip or domain with /setup eg xx.xx.xx.xx/setup to finish the install. It took me a couple of days to figure this out. I'm a complete noob, and yet to set up a stream. Good luck.

**Comment by Cedric-Demarais** [comment]
I had many pb installing it on 918+ but now i know how to install it properly with no error on synology:

- Install Docker, don't download azura's containers from synology's docker but activate SSH instead
- Connect to SSH with your admin account and next go as root with sudo -i and your admin password
- enter theses commands:
mkdir -p /var/azuracast
cd /var/azuracast
curl -fsSL https://raw.githubusercontent.com/AzuraCast/AzuraCast/master/docker.sh > docker.sh
chmod a+x docker.sh
./docker.sh install

- Cancel the installation at thirst question, you now have a azuracast.env file
- edit azuracast.env with vi (i to insert change , :w to save , :q to quit)
vi azuracast.env
Enable: ENABLE_ADVANCED_FEATURES=true
change: MYSQL_PASSWORD=toyourownpassword
set PHP_MAX_FILE_SIZE and PHP_MEMORY_LIMIT to the max file size you want to accept (250M for me)
save changes ( :w to save , :q to quit)

- Next you MUST edit the attached docker-compose.override.yml  (remove .txt at the end) with your own folders configuration.
Note 1: i am unable to have the intro.mp3 working at the moment i don't know why
Note 2: change 'myradio' in /var/azuracast/stations/myradio/ by the 'URL Stub' you WILL set on your first radio (example: radio-911) with no space, all small caps.
Note 3: with these folders mapping you can make a playlist with your recorded lives to have some replay if you want to

[docker-compose.override.yml.txt](https://github.com/AzuraCast/AzuraCast/files/5857662/docker-compose.override.yml.txt)

- Create all folders you mention in this file and put your own files at the right position (error.mp3, Favicons, etc..)

- Next i use azu-1st-install.sh (remove .txt at the end) script i made wich basically:
set COMPOSE_HTTP_TIMEOUT=300 in .env file to limit timeout errors during install , you can set it to higher value if you still have errors on second install (see below).
run installation twice with default ports in case of timeout error on slow server
run installation a third time to change default ports because there are in use on synology
change permissions of mapped folders so Azura and synology's users car read and write on it depending on their permissions

[azu-1st-install.sh.txt](https://github.com/AzuraCast/AzuraCast/files/5857595/azu-1st-install.sh.txt)

you MUST EDIT this file before running it to change folders according to your configuration and your docker-compose-override.yml file.

once edited, you can run it:
chmod +x azu-1st-install.sh
./azu-1st-install.sh

And then connect to web page on port 40080 and configure your first station.

for convenient i use also these scripts:

[azu-stop.sh.txt](https://github.com/AzuraCast/AzuraCast/files/5857700/azu-stop.sh.txt)
[azu-restart.sh.txt](https://github.com/AzuraCast/AzuraCast/files/5857701/azu-restart.sh.txt)
[azu-start.sh.txt](https://github.com/AzuraCast/AzuraCast/files/5857702/azu-start.sh.txt)

If you have subdomain or domain for your radio you can easyly configure a reverse proxy in synology's parameters to localhost port 40080 (if you use my script's settings). And in avenza enable [x] Use a web proxy for the station so everything will use this and only this port.
